Cod3rsGrowth.Web/wwwroot/webapp/controller/Detalhes.controller.js

```diff
@@ -55,19 +56,33 @@ sap.ui.define([
         aoCoincidirRota: function(evento) {
             this._exibirEspera(async () => {
                 this._carregarModelo();
-
                 const argumentos = evento.getParameter(ARGUMENTOS_DA_ROTA);
-                const clube = await ClubeServico.buscarClubePorId(argumentos.idClube);
-                let oModel = new JSONModel(clube);
-                this._modelo(NOME_MODELO_CLUBE, oModel);
+                await this._carregarClube(argumentos.idClube);
+                await this._carregarElencoLIsta();
+            });
+        },
 
-                let elenco = this._modelo(NOME_MODELO_CLUBE).getData().elenco;
-                let jogadores = await this._aoCarregarElenco(elenco);
-                oModel = new JSONModel(jogadores);
-                this._modelo(NOME_MODELO_NOME_MODELO_JOGADORES, oModel);
+        _atualizarLista: async function(){
+            this._exibirEspera(async () => {
+                const clube = this._modelo(NOME_MODELO_CLUBE).getData();
+                await this._carregarClube(clube.id);
+                await this._carregarElencoLIsta();
             });
         },
 
+        _carregarClube: async function(idClube){
+            const clube = await ClubeServico.buscarClubePorId(idClube);
+            let oModel = new JSONModel(clube);
+            this._modelo(NOME_MODELO_CLUBE, oModel);
```

Comment:
cria uma funcao tipo assim, e nao so para esse cria para todos os modelos
_modeloClube: function (modelo) {
    return this.modelo(NOME_MODELO_CLUBE, modelo);
},